pkg/controller/shared_state.go

```diff
@@ -1228,16 +1235,17 @@ func (c *SharedState) getNodesForReplica(ctx context.Context, volumeName string,
 	}
 
 	filtered := []string{}
-	numFiltered := 0
 	for _, node := range nodes {
-		if numFiltered >= numReplica {
-			break
-		}
 		if skipSet[node] {
 			continue
 		}
+		// if the node has no capacity for disk attachment, we should skip it
+		remainingCapacity, nodeExists := c.availableAttachmentsMap.Load(node)
+		if !nodeExists || remainingCapacity == nil || remainingCapacity.(*atomic.Int32).Load() <= int32(0) {
+			w.Logger().Infof("skip node(%s) because it has no capacity for disk attachment", node)
```

Comment:
This log could be quite chatty when near full attachment capacity. I would recommend a higher logging level here, e.g. `V(5)`.

Reply:
Sure. I saw that in our code base we have V(2) V(5) V(10). Does 10 mean the most important log? How to use this level to help debug?

Reply:
Lower values in verbosity level are generally more important. We have to ensure that we log enough information to be useful in production without overwhelming the logs. This is important because storage is limited, and we must support many customers and clusters. For production, we generally use verbosity level 2 or 3. For testing, we will use higher levels because the additional logging helps analyze test failures.